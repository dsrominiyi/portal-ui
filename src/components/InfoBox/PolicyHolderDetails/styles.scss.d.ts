export const infoBoxHeader: string;
export const policyHolderDetails: string;
export const header: string;
export const subHeader: string;
export const textSection: string;
