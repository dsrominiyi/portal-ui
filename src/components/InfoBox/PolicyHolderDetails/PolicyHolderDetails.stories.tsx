import React from 'react';
import { storiesOf } from '@storybook/react';
import PolicyHolderDetails, { CustomerDetails } from '.';

const stories = storiesOf('PolicyHolderDetails', module);

const testCustomerDetails = {
  name: 'Mr John Smith',
  dateOfBirth: '1987-12-09',
  address: {
    buildingNumber: '12',
    thoroughfare: 'St Peters Road',
    subThoroughfare: 'Fib Street',
    locality: 'Manchester',
    town: 'Bury',
    county: 'Greater Manchester',
    postcode: 'BL90DN'
  },
  contactDetails: [
    { type: ContactType.WorkNumber, detail: '0800000000' },
    { type: ContactType.MobileNumber, detail: '07790000000' },
    { type: ContactType.WorkEmail, detail: 'john.smith@datamatters.co.uk' }
  ]
};

const render = (
  id: string,
  isQuote: boolean,
  policyType: string,
  customerDetails: CustomerDetails
) => () => {
  return (
    <PolicyHolderDetails
      id={id}
      isQuote={isQuote}
      policyType={policyType}
      customerDetails={customerDetails}
      onClickAction={() => alert('You clicked!')}
    />
  );
};

export const storyRenders = {
  quote: render('12345', true, 'Private Car', testCustomerDetails),
  policy: render('12345', false, 'Private Car', testCustomerDetails)
};

stories.add('Quote', storyRenders.quote);
stories.add('Policy', storyRenders.policy);
