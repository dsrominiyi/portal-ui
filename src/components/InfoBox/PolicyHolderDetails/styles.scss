@import '../../../style/vars';

div.info-box-header {
  color: $brand-black;
}

.policy-holder-details {
  white-space: pre-wrap;
  font-size: 1rem;
  line-height: 1.2;
  color: $brand-charcoal;
}

.header {
  font-size: 1.5rem;
  font-weight: bold;
  color: $brand-black;
}

.sub-header {
  margin-top: 0.5rem;
  font-weight: bold;
}

.text-section {
  margin-bottom: 1rem;
}
