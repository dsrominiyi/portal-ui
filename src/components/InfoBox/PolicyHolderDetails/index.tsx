import React, { FunctionComponent, MouseEvent } from 'react';
import { utc as moment } from 'moment';

import MoreActionsLink from '../../IconLink/MoreActionsLink';
import InfoBox from '..';

import styles from './styles.scss';

export interface CustomerDetails {
  name: string;
  dateOfBirth?: string;
  address: Address;
  contactDetails: Contact[];
}

export interface Props {
  id: string;
  isQuote: boolean;
  policyType: string;
  customerDetails: CustomerDetails;
  onClickAction: (event: MouseEvent) => void;
}

const PolicyHolderDetails: FunctionComponent<Props> = ({
  id,
  isQuote,
  policyType,
  customerDetails,
  onClickAction
}: Props) => {
  const address = Object.values(customerDetails.address)
    .filter(val => val !== undefined && val !== '')
    .join(', ');

  const contacts = customerDetails.contactDetails.map((contact, index) => {
    const newLine = index !== customerDetails.contactDetails.length - 1 ? '\n' : '';
    return `${contact.type[0].toUpperCase() + contact.type.slice(1)}: ${contact.detail}${newLine}`;
  });

  const body = (
    <div className={styles.policyHolderDetails}>
      <p className={styles.header}>Policyholder</p>
      <p className={styles.subHeader}>{customerDetails.name}</p>
      {customerDetails.dateOfBirth && (
        <p className={styles.textSection}>
          {moment(customerDetails.dateOfBirth).format('DD/MM/YYYY')}
        </p>
      )}
      <p className={styles.subHeader}>Address:</p>
      <p className={styles.textSection}>{address}</p>
      <p className={styles.subHeader}>Contact:</p>
      <p className={styles.textSection}>{contacts}</p>
    </div>
  );

  return (
    <InfoBox
      header={`${isQuote ? 'Quote' : 'Policy'}: ${id}`}
      subHeader={policyType}
      moreActions={[<MoreActionsLink key="moreActionsLink" onClick={onClickAction} />]}
      items={[body]}
      additionalHeaderClasses={[styles.infoBoxHeader]}
    />
  );
};

PolicyHolderDetails.displayName = 'PolicyHolderDetails';
export default PolicyHolderDetails;
