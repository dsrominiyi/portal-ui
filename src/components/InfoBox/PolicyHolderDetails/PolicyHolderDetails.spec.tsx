import { storyRenders } from './PolicyHolderDetails.stories';

import testStorySnapshots from '../../../helpers/test/testStorySnapshots';

describe('PolicyHolderDetails', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
