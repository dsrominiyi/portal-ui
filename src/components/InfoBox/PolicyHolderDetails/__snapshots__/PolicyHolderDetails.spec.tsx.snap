// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyHolderDetails component story renders match snapshots: policy 1`] = `
<InfoBox
  additionalHeaderClasses={
    Array [
      "infoBoxHeader",
    ]
  }
  header="Policy: 12345"
  items={
    Array [
      <div
        className="policyHolderDetails"
      >
        <p
          className="header"
        >
          Policyholder
        </p>
        <p
          className="subHeader"
        >
          Mr John Smith
        </p>
        <p
          className="textSection"
        >
          09/12/1987
        </p>
        <p
          className="subHeader"
        >
          Address:
        </p>
        <p
          className="textSection"
        >
          12, St Peters Road, Fib Street, Manchester, Bury, Greater Manchester, BL90DN
        </p>
        <p
          className="subHeader"
        >
          Contact:
        </p>
        <p
          className="textSection"
        >
          Work Number: 0800000000

          Mobile Number: 07790000000

          Work Email: john.smith@datamatters.co.uk
        </p>
      </div>,
    ]
  }
  moreActions={
    Array [
      <Unknown
        onClick={[Function]}
      />,
    ]
  }
  subHeader="Private Car"
/>
`;

exports[`PolicyHolderDetails component story renders match snapshots: quote 1`] = `
<InfoBox
  additionalHeaderClasses={
    Array [
      "infoBoxHeader",
    ]
  }
  header="Quote: 12345"
  items={
    Array [
      <div
        className="policyHolderDetails"
      >
        <p
          className="header"
        >
          Policyholder
        </p>
        <p
          className="subHeader"
        >
          Mr John Smith
        </p>
        <p
          className="textSection"
        >
          09/12/1987
        </p>
        <p
          className="subHeader"
        >
          Address:
        </p>
        <p
          className="textSection"
        >
          12, St Peters Road, Fib Street, Manchester, Bury, Greater Manchester, BL90DN
        </p>
        <p
          className="subHeader"
        >
          Contact:
        </p>
        <p
          className="textSection"
        >
          Work Number: 0800000000

          Mobile Number: 07790000000

          Work Email: john.smith@datamatters.co.uk
        </p>
      </div>,
    ]
  }
  moreActions={
    Array [
      <Unknown
        onClick={[Function]}
      />,
    ]
  }
  subHeader="Private Car"
/>
`;
